afteraction-share.html

```diff
@@ -415,39 +470,88 @@ <h2>{{ page.custom_fields.share_slide_title }}</h2>
-							}
-						});
-						// if user opts not to share or has completed sharing, show donate ask
-						afteraction.on('click','.afteraction-share-ask-no, .afteraction-share-button', function(){
-							var el = $(this);
-							afteractionShowDonate(el);
-						});
-						$('.afteraction-donate-amount, .afteraction-donate-ask-yes').on('click', function(){
-							// handle adblockers preventing the loading of Google Analytics library
-							if (typeof ga !== 'undefined') {
-								ga('send', 'event', 'scrolling-signup', 'afteraction-donate-yes', window.location.pathname);
-							}
-						});
-						$('.afteraction-donate-ask-no').on('click', function(){
-							// handle adblockers preventing the loading of Google Analytics library
-							if (typeof ga !== 'undefined') {
-								ga('send', 'event', 'scrolling-signup', 'afteraction-donate-no', window.location.pathname);
-							}
-						});
+						if (afterActionNextStep === 'scroll-share') {
+							// show the share ask
+							afteractionShowShareAsk();
+							// if user opts to share, show share dialog
+							$('.afteraction-share-ask-yes').on('click', function(){
+								afteractionShowShare();
+								// handle adblockers preventing the loading of Google Analytics library
+								if (typeof ga !== 'undefined') {
+									ga('send', 'event', 'scrolling-signup', 'afteraction-share-ask-yes', window.location.pathname);
+								}
+							});
+							$('.afteraction-share-ask-no').on('click', function(){
+								// handle adblockers preventing the loading of Google Analytics library
+								if (typeof ga !== 'undefined') {
+									ga('send', 'event', 'scrolling-signup', 'afteraction-share-ask-no', window.location.pathname);
+								}
+							});
+							// if user opts not to share or has completed sharing, show donate ask
+							afteraction.on('click','.afteraction-share-ask-no, .afteraction-share-button', function(){
+								var el = $(this);
+								afteractionShowDonate(el);
+							});
+							$('.afteraction-donate-amount, .afteraction-donate-ask-yes').on('click', function(){
+								// handle adblockers preventing the loading of Google Analytics library
+								if (typeof ga !== 'undefined') {
+									ga('send', 'event', 'scrolling-signup', 'afteraction-donate-yes', window.location.pathname);
+								}
+							});
+							$('.afteraction-donate-ask-no').on('click', function(){
+								// handle adblockers preventing the loading of Google Analytics library
+								if (typeof ga !== 'undefined') {
+									ga('send', 'event', 'scrolling-signup', 'afteraction-donate-no', window.location.pathname);
+								}
+							});
+						} else if (afterActionNextStep === 'scroll-donate') {
+							// show the donate ask
+							afteractionShowDonate();
+
+							afteraction.on('click', '.afteraction-donate-ask-no', function(e){
+								// prevent getting redirected to the Get Involved URL
+								e.preventDefault();
```

Comment:
Looks like this chunk of code is repeated in above step, any way to make this more DRY?

Reply:
Hmm not sure the best way to do this without a larger refactor - maybe something we can tackle in a future sprint so as not to hold back the release this week?

Reply:
Refactoring later works for me - if we add regional variability and embedded form support in future sprints we're likely to see the logic getting more complicated, so maybe best to consider that together